### PR TITLE
Fix various bugs

### PR DESCRIPTION
**Pull request by niqore**
If a player executes /ec (Essentials) the 2 following stack traces appear. I know that a try catch isn't very clean but this is the only way I found without using NMS.

```
Caused by: java.lang.NullPointerException
	at net.minecraft.server.v1_13_R2.InventoryEnderChest.getLocation(InventoryEnderChest.java:18) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at org.bukkit.craftbukkit.v1_13_R2.inventory.CraftInventory.getLocation(CraftInventory.java:488) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at com.robomwm.prettysimpleshop.shop.ShopListener.onOpenInventory(ShopListener.java:203) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor359.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_72]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_72]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	... 26 more
```
```
Caused by: java.lang.NullPointerException
	at net.minecraft.server.v1_13_R2.InventoryEnderChest.getLocation(InventoryEnderChest.java:18) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at org.bukkit.craftbukkit.v1_13_R2.inventory.CraftInventory.getLocation(CraftInventory.java:488) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at com.robomwm.prettysimpleshop.shop.ShopListener.onClose(ShopListener.java:259) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor347.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_72]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_72]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	... 16 more
```

I also fixed this error:
```
java.lang.ClassCastException: org.bukkit.craftbukkit.v1_13_R2.block.CraftShulkerBox cannot be cast to org.bukkit.block.Chest
	at com.robomwm.prettysimpleshop.feature.ShowoffItem$2.run(ShowoffItem.java:188) ~[?:?]
	at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:76) ~[spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361) [spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:889) [spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:411) [spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:831) [spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:729) [spigot-1.13.1.jar:git-Spigot-2440e18-4228a56]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_72]
```

### CODE REVIEW

**Comment by RoboMWM** [comment]
Use three backticks on new lines to make a code block. I've fixed it for you.

**Comment by RoboMWM** [comment]
As for ender chests, I thought I already had an NPE check for this. If it's null, then it definitely is something that the plugin shouldn't care about. (And in the case of ender chests this should be the case, or at least for those opened without a corresponding block.)

It seems that spigot doesn't like this behavior - I think Paper does this properly though and returns null as necessary. getLocation should return null as per javadocs. I believe you should report this issue with spigot if these errors don't appear in paper(?)

File: src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import net.md_5.bungee.api.chat.HoverEvent;
 import net.md_5.bungee.api.chat.TextComponent;
 import net.milkbowl.vault.economy.Economy;
+
```
Comment (RoboMWM) [inline src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java]: ???

File: src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java [hunk 2]
```diff
@@ -200,8 +201,13 @@ private void onOpenInventory(InventoryOpenEvent event)
     {
         if (event.getPlayer().getType() != EntityType.PLAYER)
             return;
-        if (event.getInventory().getLocation() == null)
-            return;
+        try {
```
Comment (RoboMWM) [inline src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java]: I prefer newline braces here for consistency.

File: src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java [hunk 3]
```diff
@@ -200,8 +201,13 @@ private void onOpenInventory(InventoryOpenEvent event)
...
+        	if (event.getInventory().getLocation() == null)
+        		return;
+        }
+        catch (NullPointerException e) {
```
Comment (RoboMWM) [inline src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java:209]: Should probably add a code comment stating this is a spigot/custom inventory bug. Perhaps reference whatever bug report when you make it.

**Review by RoboMWM — CHANGES_REQUESTED** [review]
Pretty good, just some minor things. But we need to investigate why this is happening first.

**Comment by RoboMWM** [comment]
Also, try to keep each fix in a different PR. Don't lump "fixes" all in one PR.

As for the inventory thing - may also be an Essentials thing. Not sure how it's opening the enderchest to the user. Iirc slimefun didn't cause this issue with their portable ender backpacks, they just returned null for location (and was the reason why I did that null check).

File: src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java (hunk 3 above)
Comment (RoboMWM) [reply src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java:209]: Also, did you indent with tabs or something? The indentation of the return statement looks... excessive.

**Comment by RoboMWM** [comment]
Yea, supposed to return null. Not throw a NPE. https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/inventory/Inventory.html#getLocation--

I guess spigot needs to wrap their call in a try-catch?

File: src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java (hunk 1 above)
Comment (niqore) [reply src/main/java/com/robomwm/prettysimpleshop/shop/ShopListener.java]: I just tried something else before then removed an import... And git detected a difference

**Comment by RoboMWM** [comment]
Anyways, try testing Inventory#getLocation with your own plugin that opens an ender chest (or any custom inventory for that matter) and see if you also get an NPE. Would like to know if this is indeed a spigot issue.

**Review by RoboMWM — APPROVED** [review]

**Comment by RoboMWM** [comment]
@Nashoba24 have you followed up on this? 

I wonder if the enderchest thing has to do with spigot altering how custom inventories are created. I think I resolved a similar issue in another plugin, will check now.

**Comment by RoboMWM** [comment]
Ah, it probably has to do with stripping away the custom inventoryholder. https://github.com/MLG-Fortress/CustomItemRecipes/commit/eeb826cfe235ece7abe0b53cf408f123934aa0ad

Either way, spigot needs to fix this and return null instead of asking NMS and having that throw an exception.

**Comment by RoboMWM** [comment]
@Nashoba24 Please confirm if this issue still exists. If so, then please report the bug to the spigot JIRA.

**Pull request merged**